client.go

```diff
@@ -376,10 +388,20 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 			break
 		}
 	}
-
 	resp.Body = ioutil.NopCloser(bytes.NewReader([]byte{}))
 	conn.subprotocol = resp.Header.Get("Sec-Websocket-Protocol")
 
+	// Need to handle the case of there being remaining buffered bytes in the limited reader
+	if maxHeaderSizeIsSet {
+		if buffered := headerBr.Buffered(); buffered > 0 {
+			bufBytes, err := headerBr.Peek(buffered)
+			if err != nil {
+				return nil, resp, err
+			}
+			conn.br = bufio.NewReaderSize(io.MultiReader(bytes.NewReader(bufBytes), conn.br), d.ReadBufferSize)
```

Comment:
Is there a test for this statement?

Reply:
You can reperpose `TestDialRemainingHeaderBytes` with `MaxHeaderSize` to test this case.

Reply:
Thank you. This test was explicitly written to test that case but I mistakenly never changed it after removing the default value to preserve backwards compatibility in commit d79f04672a303ce06095f4ba538350dba84570ac

I've just posted #10 to fix it.